aea/test_tools/test_protocol.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2022-2023 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains test case classes based on pytest for AEA protocol testing."""
+
+
+from abc import abstractmethod
+from typing import List, Type
+from unittest import TestCase
+from unittest.mock import patch
+
+import pytest
+
+from aea.common import Address
+from aea.mail.base import Envelope
+from aea.protocols.base import Message
+from aea.protocols.dialogue.base import Dialogue, Dialogues
+
+
+class BaseMessageTestCase(TestCase):
+    """Base class to test one message encode/decode"""
+
+    __test__ = False
+
```

Comment:
why not pytest-style? May be good if we use a particular style consistently, whichever it turns out to be. 
`__test__` would not be needed either than

Reply:
@Karrenbelt 
cause this
```
def test_test_message_case():
    """Test test cases run."""
    suite = unittest.defaultTestLoader.loadTestsFromTestCase(TestMessages)
    result = unittest.TextTestRunner().run(suite)
    assert result.errors == []
    assert result.failures == []
    assert result.wasSuccessful()

    class FailTestMessages(TestMessages):
        """Base class to test message construction for the protocol."""

        def build_inconsistent(self) -> List[Message]:
            """Build inconsistent messages to be used for testing."""
            return [
                TProtocolMessage(
                    performative=TProtocolMessage.Performative.PERFORMATIVE_CT,
                    content_ct=custom_data_model,
                ),
            ]

    suite = unittest.defaultTestLoader.loadTestsFromTestCase(FailTestMessages)

    result = unittest.TextTestRunner().run(suite)
    assert result.errors == []
    assert not result.wasSuccessful()
    assert len(result.failures) == 1
    assert "test_messages_inconsistent" in str(result.failures)

```

Reply:
got code for unitest, so this is why applied unittest.TestCase